docs/developer-docs/defi/token_integrations/index.mdx

```diff
@@ -0,0 +1,95 @@
+
+## Direct integration
+Direct integration is supported by libraries for Rust and Typescript, but proficient use of these tools would require familiarity with the following concepts:
+
+
+Relevant aspects of execution semantics of ICP messages:
+- The distinction between update calls and querry calls
+- Reading the result of update calls
+- Deduplication guarantees of the internet computer protocol
+- Ingress message lifecycle 
+- How to verify the result of calls to canisters
+- API of the token canister(s) & their semantics
+- Deduplication guarantees by the ledger 
+- How to fetch the token ledger & verify it is genuine
+- CBOR (Concise Binary Object Representation)
+- Candid (ICP’s interface description language)
+
+### Calling a method on an ICP canister 
+A detailed description of how to use the API of the Internet Computer is included in this [section](#http-call-overview) of the Internet Computer Specification. Here we provide an overview:
+Calling canister methods is done by constructing a message which specifies among other things the canister, the method and its parameters, and submitting this message to the network.  
+Depending on how they are submitted, these calls are executed either replicated (by all nodes in the subnet) or unreplicated (only by one node in the subnet).  This represents a trade-off. Replicated calls are secure because tampering with the result requires tampering with a majority of nodes in the subnet. We call these “update calls”. However, replicated calls are slower since they involve running a consensus protocol on the input, and certifying the output. These calls are asynchronous.  The call is submitted to the Internet Computer and the result can only be read, by making an additional call, after the Internet Computer has certified the output. 
+Non-replicated calls or “query calls” are executed by a single node in the network. Execution can therefore be synchronous, thus much faster. Because a single node is involved in the execution, the result should not be trusted (since the node can be malicious). Nonetheless, one can build trust in the results of query calls via certificates (signatures) which the ICP network can produce. For example, the hash of the last block in the ICP ledger comes with such a certificate which can be verified using the public key of the ICP network. Together with checking the correct chaining of transactions within the ICP ledger, this guarantees that the chain of transactions fetched via queries is indeed genuine. The details of how the certificates are produced and can be verified are in this companion document [here](/docs/references/http-gateway-protocol-spec). 
```

Comment:
```suggestion

```

Reply:
It is important to point out how the IC works in order to understand why direct integration works the way that it does and why it is much more complex than using rosetta. I wouldn't take expect users that read this guide to be very familiar with the consensus and replication mechanism of the IC.

Reply:
Could we link to the other docs that cover this then? Ideally, we want as little duplication across the docs as possible.